src/itemCheckbox/ItemCheckbox.unit.tsx

```diff
@@ -1,76 +1,51 @@
+  name: 'name',
+  className: 'custom-class-name',
+  labelTitle: 'Label title',
+  data: 'Data',
+  dataInfo: 'Data info',
+  checked: false,
+  disabled: false,
+  onChange() {},
+  status: ItemCheckboxStatus.DEFAULT,
+  key: 0,
+}
 
-  it('Should use the Item component', () => {
-    const itemCheckbox = shallow(<ItemCheckbox {...defaultProps} />)
-    expect(itemCheckbox.exists()).toBe(true)
-  })
-  it('Should forward its props to the Item component', () => {
-    const itemCheckbox = renderer.create(<ItemCheckbox {...defaultProps} />).toJSON()
-    expect(itemCheckbox).toMatchSnapshot()
-  })
-  it('Should display a CheckIcon when the input is checked', () => {
-    const itemCheckbox = renderer.create(<ItemCheckbox {...defaultProps} checked />).toJSON()
-    expect(itemCheckbox).toMatchSnapshot()
-  })
-  it('Should display a Loader when the component is in loading status', () => {
-    const itemCheckbox = renderer
-      .create(<ItemCheckbox {...defaultProps} status={ItemCheckboxStatus.LOADING} />)
-      .toJSON()
-    expect(itemCheckbox).toMatchSnapshot()
-  })
-  it('Should bind the onChange callback of the input', () => {
-    const itemCheckbox = mount(<ItemCheckbox {...defaultProps} />)
-    expect(itemCheckbox.find('input').prop('onChange')).toEqual(itemCheckbox.instance().onChange)
-  })
-  it('Should call the onChange prop with name and value when the input changes (check)', () => {
-    const onChangeMock = jest.fn()
-    const itemCheckbox = shallow(<ItemCheckbox {...defaultProps} onChange={onChangeMock} />)
-    itemCheckbox.instance().onChange()
-    expect(onChangeMock).toHaveBeenCalledWith({
-      name: defaultProps.name,
-      value: true,
+function createProps(props: Partial<ItemCheckboxProps> = {}): ItemCheckboxProps {
+  return { ...defaultProps, ...props }
+}
+
+describe('ItemCheckbox', () => {
+  it('Should have a label around the input', () => {
+    const props = createProps({
+      label: 'This is the field label',
+      labelTitle: '',
+      data: '',
+      dataInfo: '',
     })
+    render(<ItemCheckbox {...props} />)
+    expect(
+      screen.getByLabelText('This is the field label', { selector: 'input[type="checkbox"]' }),
+    ).toBeInTheDocument()
   })
-  it('Should handle disabled prop', () => {
-    const itemCheckbox = mount(<ItemCheckbox {...defaultProps} />)
-    expect(itemCheckbox.prop('disabled')).toBeFalsy()
 
-    itemCheckbox.setProps({ disabled: true })
-    expect(itemCheckbox.prop('disabled')).toBeTruthy()
-  })
-  it('Should call the onChange prop with name and value when the input changes (uncheck)', () => {
-    const onChangeMock = jest.fn()
-    const itemCheckbox = shallow(<ItemCheckbox {...defaultProps} onChange={onChangeMock} checked />)
-    itemCheckbox.instance().onChange()
-    expect(onChangeMock).toHaveBeenCalledWith({
-      name: defaultProps.name,
-      value: false,
-    })
+  it('Should display a Loader when the component is in loading status', () => {
```

Comment:
This test doesn't assert on the visibility of a loader 🤔

Reply:
Oops, good catch! I think I just forgot to write this test 😅

Reply:
Updated